modules/router-store/src/init_router_store.ts

```diff
@@ -0,0 +1,17 @@
+import { ENVIRONMENT_INITIALIZER, inject, Provider } from '@angular/core';
+import { StoreRouterConnectingService } from './store_router_connecting.service';
+
+/**
+ * Initializer for the router-store that ensures the initialization logic is run alongside environment
+ * initialization.
+ */
+export const _initRouterStore: Provider[] = [
+  {
+    provide: ENVIRONMENT_INITIALIZER,
+    multi: true,
+    useValue() {
+      inject(StoreRouterConnectingService);
+    },
```

Comment:
This should be useFactory and return a callback function

```suggestion
    useFactory() {
      return () => {
        inject(StoreRouterConnectingService);
      };
    },
```

Reply:
Does the env initializer token need it to be a factory? Cause all of the tests passed so I think useValue worked?